packages/app/src/systems/Assets/components/AssetLogo/AssetLogo.tsx

```diff
@@ -0,0 +1,46 @@
+import { Image, Avatar } from '@fuel-ui/react';
+import { useMemo } from 'react';
+
+import type { Asset } from '../../services';
+import { getAssetEth, getAssetFuel } from '../../utils';
+
+type EthAssetLogoProps = {
+  asset?: Asset;
+  size?: number;
+  alt?: string;
+};
+
+const DEFAULT_SIZE = 18;
+
+export const AssetLogo = ({
+  alt = 'AssetLogo',
+  asset,
+  size = DEFAULT_SIZE,
+}: EthAssetLogoProps) => {
+  const { image, address } = useMemo(() => {
+    if (asset?.icon) {
+      return { image: asset.icon };
+    }
+
+    if (!asset) return {};
+
+    const ethAsset = getAssetEth(asset);
+    const fuelAsset = getAssetFuel(asset);
+
+    return {
+      address: ethAsset?.address || fuelAsset?.assetId || fuelAsset?.contractId,
+    };
+  }, [asset?.symbol]);
+
+  if (!image && !address) return null;
```

Comment:
Why do you need this check here? On the return you already have the check for the image and the replace for the address.

And I think is better to show the generated and after replace with the image them return null.